swap/swap.go

```diff
@@ -318,7 +317,7 @@ func (s *Swap) handleEmitChequeMsg(ctx context.Context, p *Peer, msg *EmitCheque
 		return err
 	}
 
-	otherSwap, err := contract.InstanceAt(cheque.Contract, s.backend)
+	contract, err := swapcontract.InstanceAt(cheque.ContractAddress, s.backend)
```

Comment:
same situation here: we definitely should change `otherSwap`, but using just `contract` is what we are trying to avoid.

my suggestions: `otherChequebook`, `chequebookContract`, or similar